Client/Frontend/Browser/User Scripts/ScriptFactory.swift

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import WebKit
+
+/// An error representing failures in loading scripts
+enum ScriptLoadFailure: Error {
+  case notFound
+}
+
+/// A class that helps in the aid of creating and caching scripts.
+class ScriptFactory {
+  /// A shared instance to be shared throughout the app.
+  ///
+  /// - Note: Perhaps we will move this to the tab to be managed on a tab basis.
+  static let shared = ScriptFactory()
+  
+  /// Ensures that the message handlers cannot be invoked by the page scripts
+  public static let messageHandlerTokenString: String = {
+    return UUID().uuidString.replacingOccurrences(of: "-", with: "", options: .literal)
+  }()
+  
+  /// This contains cahced script sources for a script type. Avoids reading from disk.
+  private var cachedScriptSources: [ScriptSourceType: String]
+  
+  /// This contains cached altered scripts that are ready to be inected. Avoids replacing strings.
+  private var cachedDomainScriptsSources: [UserScriptType: WKUserScript]
+  
+  init() {
+    cachedScriptSources = [:]
+    cachedDomainScriptsSources = [:]
+  }
+  
+  /// Clear some caches in case we need to.
+  ///
+  /// Should only really be called in a memory warning scenario.
+  func clearCaches() {
+    cachedScriptSources = [:]
+    cachedDomainScriptsSources = [:]
+  }
+  
+  /// Returns a script source by loading a file or returning cached data
+  private func makeScriptSource(of type: ScriptSourceType) throws -> String {
+    if let source = cachedScriptSources[type] {
+      return source
+    } else {
+      let source = try type.loadScript()
+      cachedScriptSources[type] = source
+      return source
+    }
+  }
+  
+  /// Get a script for the `UserScriptType`.
+  ///
+  /// Scripts can be cached on two levels:
+  /// - On the unmodified source file (per `ScriptSourceType`)
+  /// - On the modfied source file (per `UserScriptType`)
+  func makeScript(for domainType: UserScriptType) throws -> WKUserScript {
+    var source = try makeScriptSource(of: domainType.sourceType)
+
+    // First check for and return cached value
+    if let script = cachedDomainScriptsSources[domainType] {
+      return script
+    }
+    
+    switch domainType {
+    case .farblingProtection(let etld):
+      let randomConfiguration = RandomConfiguration(etld: etld)
+      let fakeParams = try FarblingProtectionHelper.makeFarblingParams(from: randomConfiguration)
+
+      #if DEBUG
+      print("[ScriptFactory] eTLD+1: \(etld)")
+      print("[ScriptFactory] Seed:   \(randomConfiguration.seed)")
+      print("[ScriptFactory] Params: \(fakeParams)")
+      #endif
+      
+      source = "\(source)\n(\(String(describing: fakeParams)))"
```

Comment:
No need for `String(describing:...)` because `fakeParams` is already a string, so pass it directly.